crates/runc-shim/src/service.rs

```diff
@@ -79,7 +79,20 @@ impl Shim for Service {
             None => {}
         }
 
-        let address = spawn(opts, &grouping, Vec::new()).await?;
+        let mut vars: Vec<(&str, &str)> = Vec::new();
+        #[cfg(target_os = "linux")]
+        let mut thp_disabled = String::new();
+        #[cfg(not(target_os = "linux"))]
```

Comment:
```rust
        let disabled = if cfg!(target_os = "linux") {
            if let Ok(x) = prctl::get_thp_disable() {
                let _ = prctl::set_thp_disable(true);
                true
            } else {
                false
            }
        } else {
            false
        };

        let vars = vec![("THP_DISABLED", &disabled.to_string())]
```
?

Reply:
since it sounds like  `#[cfg(target_os = "linux")]` is needed, it still might be clearer to do something like:

```rust
 let thp_disabled = false;
#[cfg(target_os = "linux")]
let thp_disabled = match prctl::get_thp_disable() {
    Ok(x) => {
        let _ = prctl::set_thp_disable(true);
        true
    }
    Err(_) => false,
};
let vars: Vec<(&str, &str)> = vec![("THP_DISABLED",  &disabled.to_string())];
```

Reply:
Here we need the return value of the ```prctl::get_thp_disable``` function and assign this return value to the variable thp_disable. That is, the ```x``` in ```Ok(x)``` is needed, not just judging whether it is ```Ok()``` or ```Err()```. This value means the state before setting the set_thp_disable, and will be used to ```set_thp_disable``` before starting runc later.

Reply:
> Here we need the return value of the `prctl::get_thp_disable` function and assign this return value to the variable thp_disable. That is, the `x` in `Ok(x)` is needed, besides, `x` may be true or false. We should not just judge whether it is `Ok()` or `Err()`.

`get_thp_disable` returns a `Result<bool, i32>` so we could still return `x` instead of converting to string here. using the `bool` makes it clearer in the code than `x.tostring` and `string::new()` which is semantically unclear what `string:new()` is in this case.

Reply:
There are actually 3 states here, true, false and error. When the get_thp_disable function returns error, it means that the thp parameter cannot be obtained, if the set_thp_disable is executed at this time, it will cause runc to be unable to recover the value of thp disable, therefore, thp_disabled parameter needs 3 states, true, false, error, due to the variable life cycle, so string is used here to return.